packages/block-editor/src/components/heading-level-control/index.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Dropdown,
+	Toolbar,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { DOWN } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import HeadingLevelIcon from './heading-level-icon';
+
+const HEADING_LEVELS = [ 1, 2, 3, 4, 5, 6 ];
+const POPOVER_PROPS = {
+	className: 'block-editor-heading-level-dropdown',
+	position: 'bottom right',
+	isAlternate: true,
+};
+
+/** @typedef {import('@wordpress/element').WPComponent} WPComponent */
+
+/**
+ * HeadingLevelControl props.
+ *
+ * @typedef WPHeadingLevelControlProps
+ *
+ * @property {number} selectedLevel The chosen heading level.
+ * @property {(newValue:number)=>any} onChange Callback to run when toolbar value is changed.
+ * @property {boolean} isParagraphAllowed Append paragraph option with zero level to the available levels.
+ */
+
+/**
+ * The `HeadingLevelControl` component renders a dropdown menu that displays
+ * heading level options for the selected block. This component is used to set
+ * the level of the heading in a block and also allows to use a `paragraph`
+ * element by setting the level to zero (`0`).
+ * This would also need handling in the block's `edit` function.
```

Comment:
Same thought as my previous comment. We might be able to provide additional clarity here.  
```suggestion
 * element by setting the level to zero (`0`). The block's `edit` function would
 * also need to handle the zero value.
```